### PR TITLE
refactor(unlock-app): Refactor pricing code into hooks

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -13050,7 +13050,7 @@ __metadata:
   languageName: node
   linkType: hard
 
-"@types/node@npm:*, @types/node@npm:>=12.0.0, @types/node@npm:>=6, @types/node@npm:>=8.1.0":
```
Comment (julien51) [inline yarn.lock:13053]: Is this yarn file up to date?

Comment (searchableguy) [reply yarn.lock:13053]: Yes

File: unlock-app/src/components/interface/checkout/main/Confirm.tsx [hunk 2]
```diff
@@ -659,17 +587,17 @@ export function Confirm({
...
+                        {item.amount < Number(lock!.keyPrice) ? (
                           <Badge variant="green" size="tiny">
                             {discount}% Discount
                           </Badge>
                         ) : null}
                       </div>
 
                       <div className="font-bold">
-                        {item.amount === '0'
+                        {item.amount <= 0
```
Comment (julien51) [inline unlock-app/src/components/interface/checkout/main/Confirm.tsx:598]: pfew. Hopefully we never have less than 0 :D

**Review by julien51 — APPROVED** [review]
LGTM!

**Pull request merged**